### PR TITLE
feat(Target):  Pass through "background_page" types (#2439)

### PR DESCRIPTION
**Pull request by Runaider**

### CODE REVIEW

**Comment by Runaider** [comment]
I signed it!

**Review by ebidel — APPROVED** [review]

File: test/target.spec.js [hunk 1]
```diff
@@ -36,6 +36,12 @@ module.exports.addTests = function({testRunner, expect}) {
       expect(allPages).toContain(page);
       expect(allPages[0]).not.toBe(allPages[1]);
     });
+    it('should allow background_page target type to pass through', async({browser}) => {
+      const targets = browser.targets();
+      targets[0]._targetInfo.type = 'background_page';
```
Comment (aslushnikov) [inline test/target.spec.js]: let's be real: can you please create a minimalistic chrome extension under `/test/assets/simple-extension` and load it in the headful browser?

Comment (Runaider) [reply test/target.spec.js]: Yes, will do it later today

File: docs/api.md [hunk 2]
```diff
@@ -2595,7 +2595,7 @@ If the target is not of type `"page"`, returns `null`.
 #### target.type()
 - returns: <[string]>
 
-Identifies what kind of target this is. Can be `"page"`, `"service_worker"`, `"browser"` or `"other"`.
```
Comment (JoelEinbinder) [inline docs/api.md:2598]: As this list grows, we should probably provide some explaining links as to what these types are: [`"background_page"`](https://developer.chrome.com/extensions/background_pages)

**Comment by JoelEinbinder** [comment]
Thanks for the patch, looks good!

File: test/target.spec.js (hunk 1 above)
Comment (Runaider) [reply test/target.spec.js]: I have updated the test to use a headfull browser with a bare extension. I'm not sure about starting a new browser instance in a spec that already accepts a headless browser instance, Should I leave the test where it is or move it to another spec?

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**